### PR TITLE
[v1.1] Add github issue to pr links and PR template for better documents

### PR DESCRIPTION
**Pull request by xuechendi**

### CODE REVIEW

**Comment by xuechendi** [comment]
What is expected to see after this PR:
image
image

**Comment by xuechendi** [comment]
and a new link to issue will be created
image

**Comment by xuechendi** [comment]
@Jian-Zhang , I added some PR and issue support in this PR, please take a review, so after this PR.
Any one send new PR would see a PR template indicates him/her what is the PR title and what is PR format.
And issue and PR will be automatically linked with PR title.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+-->
+
+
+### Does this PR introduce _any_ user-facing change?
+<!--
+If yes, please clarify the change.
+-->
+
+
+### How was this patch tested?
```
Comment (Jian-Zhang) [inline .github/pull_request_template.md:33]: assuming most of the PR will be covered by github workflows automatically, do we need this?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+### Why are the changes needed?
+<!--
+Please clarify why the changes are needed. For instance,
+  1. If you propose a new API, clarify the use case for a new API.
+  2. If you fix a bug, you can clarify why it is a bug.
+-->
+
+
+### Does this PR introduce _any_ user-facing change?
```
Comment (Jian-Zhang) [inline .github/pull_request_template.md:27]: We probably don't need this.

**Review by Jian-Zhang — COMMENTED** [review]
Thanks, this is pretty helpful. Please check my comments.

File: .github/pull_request_template.md (hunk 1 above)
Comment (xuechendi) [reply .github/pull_request_template.md:33]: @Jian-Zhang , by adding this, we can make sure all path will be tested. Committer can either point to a pre-defined UT or add new UT to cover this new PR, Or simply say this is document fix, no tests required.
So we won't add new function out UT coverage.

File: .github/pull_request_template.md (hunk 2 above)
Comment (xuechendi) [reply .github/pull_request_template.md:27]: I'll remove this

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Please clarify why the changes are needed. For instance,
+  1. If you propose a new feature/API, clarify the use case for a new feature/API.
+  2. If you fix a bug, mark which bug being fixed.
+-->
+
+
+### How was this patch tested?
+<!--
+please point to existing unittest or add new unittest to cover your changes. If this is a document change, please skip this request. 
+-->
```
Comment (xuechendi) [inline .github/pull_request_template.md:30]: @Jian-Zhang , I changed the description here

**Review by Jian-Zhang — APPROVED** [review]

**Pull request merged**